### ISSUE TITLE
[dev.icinga.com #11309] Allow to bulk execute monitoring actions based on the current filter without need to list or select objects

### ISSUE THREAD

**Issue by icinga-migration** [Icinga/icingaweb2#2344]
This issue has been migrated from Redmine: https://dev.icinga.com/issues/11309

**Created by koudis on 2016-03-04 14:43:16 +00:00**

Assignee: _elippmann_
Status: _Feedback_ 
Target Version: _Backlog_
Last Update: _2016-12-15 17:17:08 +00:00 (in Redmine)_

---

Hi,
I would like to be able to to set schedule downtime or acknowledge based on filter without need to list or select objects.

Scenario:
In Service Problems I will add some filter setting and create specific filter with 5k+ services in it. I want to do action on all of them without the need to view and select all services.

**Changesets**

_2016-12-15 17:14:01 +00:00_ by _(unknown)_ a88822cb1479606174e65c108e576c3d67198410

    Add links in hosts and services lists to show all listed objects
    
    This allows executing bulk actions without having selected all of them manually.
    
    refs #11309

---

**Relations**:

* duplicates [#11309](https://dev.icinga.com/issues/11309)
* blocks [#8584](https://dev.icinga.com/issues/8584)

**Comment by helmo** [comment]
This would be really useful ... In Nagios I frequently used  the 
'Schedule a check of all services on this host' and 'schedule downtime for all services on this host'  links.

That;s something I've not been able to find in Icingaweb2

**Comment by trefzer** [comment]
I use this functionality very often. As a workaround I created a dummy host (always up) per hostgroup and depend all hosts in the group on it. If you schedule a downtime of the dummy host, you can inherit to all depend hosts !

**Comment by Punkoivan** [comment]
We're also interested in such feature.
@trefzer  thank you for your dirty hack, I think we should give it a try! :) 

also we will provide some python script to downtime services in hostgroup, once done I'll ask my manager to share with community.

**Comment by mkornuser** [comment]
Me2 ;-) 
In Classic Interface we could apply Operations to Host- and Service-Groups. Host- and Service-Groups can be regarded as permanent Filters, so their Content is kinda static.
E.g. one repetitative common Task is to schedule Downtime to all Services in a Service-Group.
Setting up a Filter and saving this Filter for later use is more expensive than using a Service-Group, as usually it already exists.

**Comment by mwaldmueller** [comment]
ref/NC/629029

**Comment by Tqnsls** [comment]
Hi, we have several hostgroups matching different corporation-sites or functionalities and patching-operations.
icinga Infrastructure: about 5k hosts and 55k services
We are really interested in having such features like setting a given filter in downtime or a given hostgroup.

I remember that icinga1 had these commands "SCHEDULE_HOSTGROUP_SVC_DOWNTIME" "SCHEDULE_HOSTGROUP_HOST_DOWNTIME" that one could use to set the hosts or services of a hostgroup in downtime. 
I just checked that these commands also work in the icinga2 commandfile.
would it not be possible to implement it in the web gui?

**Comment by nilmerg** [comment]
It's part of Icinga DB Web. Chances that this will be introduced in the monitoring module are pretty low.

**Comment by Tqnsls** [comment]
Yes sure, but I thought that if the commands are implemented in the commandfile it's possible that you can implement something like a checkbox or another option for hostgroup-downtimes.

**Comment by nilmerg** [comment]
In case I wasn't clear enough, Icinga DB Web supports sending commands (over the Icinga 2 API) based on a filter. Basically any filtered list [allows to run commands on its results](https://github.com/Icinga/icingadb-web#bulk-operations).

The monitoring module of Icinga Web 2 does not support that nor are there any plans to implement this feature since Icinga DB Web has it.

The command options you speak of are indeed supported by Icinga 2, but the entire [external command listener](https://icinga.com/docs/icinga2/latest/doc/09-object-types/#externalcommandlistener) is deprecated. We are not going to add features to our current implementation for that anymore.

**Comment by adaevitncar** [comment]
Will this issue be closed? Nothing happened for several years and the issue is still open.

**Comment by lippserd** [comment]
This is possible with Icinga DB Web: https://github.com/Icinga/icingadb-web?tab=readme-ov-file#bulk-operations
